app/shifts_controller.py

```diff
@@ -216,9 +218,11 @@ def shift_processor(self):
         hd_shifts = refresh_shifts(gsheet_sd_schedule)
 
         scan_row = 0  # tracks the correct row number for scanner time in and out
+        last_shift = 0  # tracks the last shift to be analyzed successfully
         shift_count = -1  # tracks the current part of the extra shift(s)
         lower_bound = -1
         upper_bound = -1
+        name_present = 0  # set to 0, this means the student's name in hd_shift is present within the scan_shift data
```

Comment:
The name_present variable could use some work I think. Your comments are fantastic! But it might be better if the variable name and values are more descriptive.

For example, the values 0, 1, and 2 don't mean anything. Maybe use text to represent those values.